hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/metadata/SparkHoodieBackedTableMetadataWriter.java

```diff
@@ -159,6 +162,13 @@ protected void commit(String instantTime, Map<MetadataPartitionType, HoodieData<
         compactIfNecessary(writeClient, instantTime);
       }
 
+      // It is possible that the given instantTime already exists in metadata table,
```

Comment:
@codope Can you please explain this case?

The new partition initialization should use a unique timestamp (with a suffix) and not conflict with any existing deltacommit. 

If two commits in dataset are attached to same deltacommit then it may cause issues with the log block reading and rollbacks/ restore functionality. 

Please also check https://github.com/apache/hudi/pull/8684 where the new partition enabling has been changed to:
1. Use bulkInsert for initial commit
2. Always use a unique timestamp on MDT

Reply:
This came up while I was testing a scenario where record index was enabled after a few commits. It might be fixed by #8684. I'll rebase and check again.

Reply:
Working for the same scenario now. Can you please rebase?